### DISCUSSION TITLE
List of Stochastic Optimizers

### DISCUSSION THREAD

**Discussion by hwborchers** [SudoWodo/StochOptim#28] [General]
This is a list of possible stochastic optimizers found in different R packages, together with some of their main options. I ask to suggest more of these or to suggest which ones we should better not include.

    | Package  | Function   | popsize        | maxiter  | tolerance | trace       |
    |----------|------------|----------------|----------|-----------|-------------|
    | DEoptim  | DEoptim    | NP             | itermax  | reltol    | trace       |
    | DEoptimR | JDEoptim   | NP             | maxiter  | tol       | trace       |
    | RcppDE   | DEoptim    | NP             | itermax  | reltol    | trace       |
    | NMOF     | DEopt      | nP             | -        | -         | printDetail |
    | adagio   | simpleDE   | n              | nmax     | -         | log         |
    | GenSA    | GenSA      | -              | maxit    | -         | verbose     |
    | NMOF     | SAopt      | -              | nS       | -         | printDetail |
    | GA       | ga         | popSize        | maxiter  | -         | -           |
    | NMOF     | GAopt      | nP             | -        | -         | printDetail |
    | PSO      | psoptim    | s              | maxit    | abstol    | trace       |
    | NMOF     | PSopt      | nP             | -        | -         | printDetail |
    | adagio   | pureCMAES  | -              | stopeval | -         | -           |
    | parma    | cmaes      | PopSize        | maxIter  | TolX      | -           |
    | cmaes    | cma_es     | mu             | maxit    | -         |             |
    | cmaesr   | cmaes      | mu             | -        | -         | monitor     |
    | rCMA     | cmaOptimDP | -              | -        | -         | verbose     |
    | RCEIM    | ceimOpt    | Ntot           | maxIter  | epsilon   | verbose     |
    | nloptr   | isres      | pop.size       | maxeval  | xtol_rel  | nl.info     |
    | nloptr   | stogo      | -              | maxeval  | xtol_rel  | nl.info     |
    | soma     | soma       | populationSize | -        | -         | -           |
    |----------|------------|----------------|----------|-----------|-------------||

**Comment by hwborchers** [comment]
Here is a possible result for the ModLangerman function:

    G = sopm(par = par, fn = fn, lower = bounds$lower, upper = bounds$upper,
             methods = "ALL", maxiter = 5000, popsize = 2000,
             excl = c("hydropso", "cmaoptim", "smco"))

    tibble::as_tibble(G[1:20, ])
    ## # A tibble: 20 × 3
    ##   method         fmin   time
    ##   <chr>         <dbl>  <dbl>
    ## 1 deopt      -0.965   21.7  
    ## 2 crs2lm     -0.965    1.22
    ## 3 simplede   -0.965   34.7  
    ## 4 cppdeoptim -0.959    5.06
    ## 5 malschains -0.517   21.3  
    ## 6 deoptimr   -0.517   10.7  
    ## 7 genoud     -0.513    3.22
    ## 8 ga         -0.513   24.7  
    ## 9 deoptim    -0.513   11.0  
    ## 0 purecmaes  -0.483    4.70
    ## 1 gensa      -0.275    5.52
    ## 2 isres      -0.275    1.50
    ## 3 pso        -0.1      2.90
    ## 4 psopt      -0.1     21.6  
    ## 5 ceimopt    -0.0896  12.4  
    ## 6 simpleea   -0.0715   0.730
    ## 7 soma       -0.0713   0.245
    ## 8 cmaes      -0.0532   6.91
    ## 9 cmaesr     -0.0356   1.21

Here `methods="ALL"` means to apply all methods known to `sopm()`, and `excl=c(...)` means to exclude some of those (instead of listing all methods you would like to apply).

**Reply by SudoWodo** [reply]
Will add `methods="ALL"`  and `excl=c(...)` quickly.

**Reply by SudoWodo** [reply]
Should maxiter and popsize be passed as a parameter `maxiter = 5000, popsize = 2000` or passed in a control list like :-
`control = list(maxiter = 5000, popsize = 2000)` ?

**Comment by nashjc** [comment]
My choice would be via the control list. But that is just 1 opinion.

J
